Elevato/Lift.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Elevato.Models;
+using Elevator.Extand;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elevator
+{
+    public class Lift : ILift
+    {
+        private int _CurrentFloor;
+        private readonly List<int> _path;
+
+        public volatile bool IsMoving;
+        public volatile Direction direction;
+
+        public EventHandler<int> IsMoved;
+
+        public Lift()
+        {
+            _CurrentFloor = 3;
+            _path = new List<int>();
+        }
+
+        public void Request(Message message)
+        {
+            if (_path.Contains(message.FloorNumber))
+            {
+                if (message.Direction == direction.ToString())
+                    _path.Insert(_path.IndexOf(message.FloorNumber), message.FloorNumber);
+            }
+            else
+                AddStop(message);
+        }
+
+        private void AddStop(Message message)
+        {
+            var currentFloor = CalculateCurrentFloor();
+            if (message.FloorNumber > currentFloor)
+                for (var i = currentFloor; i <= message.FloorNumber; i++)
+                    _path.Add(i);
+            else
+                for (var i = currentFloor; i >= message.FloorNumber; i--)
+                    _path.Add(i);
+            // this one for reaching the destination
+            _path.Add(message.FloorNumber);
+        }
+
```

Comment:
Private method - this should be in another class and made public

Reply:
but I don't want this method get accessed by other place in my code

Reply:
Private methods should be avoided... 
if you move it into its own class you can still control access to it, nothing will be able to access it if that class is not initialized